src/loader/elf.rs

```diff
@@ -22,6 +24,16 @@ pub const ELFMAG0: ::std::os::raw::c_uint = 127;
 pub const ELFDATA2LSB: ::std::os::raw::c_uint = 1;
 pub const PT_LOAD: ::std::os::raw::c_uint = 1;
 
+// ELF Note p_type used to parse PVH entry point address
```

Comment:
(Note that if you end up restructuring like this, `bootparam.rs` needs to suffer the same fate.)

Reply:
> This file is autogenerated by `bindgen`. I'd prefer avoiding to edit it directly. How about renaming the original `elf.rs` to `elf_gen.rs` or in a different `elf_gen` module altogether, and adding the PVH-specific definitions in a standalone, non-generated file?

My argument for modifying this file and not creating new ones is because `elf.rs` was autogenerated by `bindgen` but it seems to have been manually (and heavily) edited before arriving at this current version. 
Checking out upstream linux commit `806276b7f07a39a1cc3f38bb1ef5c573d4594a38` and looking at `linux include/uapi/linux/elf.h` shows that PT_NOTE and Note header definitions were already present in the upstream source when `elf.rs` was generated by `bindgen`. These definitions and many others were probably pruned to keep the file small since they were not needed at the time, but from a logical standpoint they should live in this file.

If you agree with my argument above, then perhaps the better solution is to run `bindgen` again on `include/uapi/linux/elf.h`, but I think that is out of the scope for this specific PR. If you don't agree, please let me know and I'll create a new `elfnote.rs` to contain the Note definitions I am adding.

Just to clarify, I did include the `pub const XEN_ELFNOTE_PHYS32_ENTRY: u32 = 18;` definition here that comes from a different header (`include/xen/interface/elfnote.h`) since I couldn't think of a better place for it and thought that it fit in the same scope as the Note definitions.

Reply:
Yap, I manually edited this file because we didn't need all those defines. I am fine with manually editing it actually and only include what we need in it. Thoughts?

Reply:
> I am fine with manually editing it actually and only include what we need in it. 

I'm OK with deleting unnecessary code but not with adding new code which would be lost when regenerating the file.

> perhaps the better solution is to run `bindgen` again on `include/uapi/linux/elf.h`

Agreed.

> but I think that is out of the scope for this specific PR. If you don't agree, please let me know and I'll create a new `elfnote.rs` to contain the Note definitions I am adding.

I'd be OK with leaving them in if the naming was consistent with what `bindgen` produces. IMO you could copy-paste the relevant definitions and tests instead of manually defining them. I ran `bindgen` and it looks like the corresponding struct is `elf64_note`?

>  I did include the` pub const XEN_ELFNOTE_PHYS32_ENTRY: u32 = 18;` definition here that comes from a different header
> That should be put somewhere else since it will be clobbered if bindgen is run again.

Agreed.

Reply:
Please see last revision, specifically the `pvh: Add ELF Note definitions` commit.

I ran `bindgen` again on `include/uapi/linux/elf.h` and did lots of manual editing. The resulting file is mostly the same, with some minor differences that are likely due to me using a different set of options when invoking bindgen, or to changes to bindgen algorithm since the original `elf.rs` was generated. Nothing too disruptive I think. e.g. constants like `EI_MAG0` are now defined using Rust primitive types (`u32`) as opposed to the C-equivalent types (`::std::os::raw::c_uint`) and  the Clone trait is now automatically derived.

I moved the definition of `XEN_ELFNOTE_PHYS32_ENTRY: u32 = 18;` to the `parse_elf_note()` helper since that is the only place that needs it.